manager/container_test.go

```diff
@@ -319,3 +319,19 @@ func TestOnDemandHousekeepingReturnsAfterStopped(t *testing.T) {
 
 	mockHandler.AssertExpectations(t)
 }
+
+var psOutput = [][]byte{
+	[]byte("root       15886       2 23:51  0.1  0.0     0      0 I    00:00:00 kworker/u8:3-ev   3 -\nroot       15887       2 23:51  0.0  0.0     0      0 I<   00:00:00 kworker/1:2H      1 -\nubuntu     15888    1804 23:51  0.0  0.0  2832  10176 R+   00:00:00 ps                1 8:devices:/user.slice,6:pids:/user.slice/user-1000.slice/session-3.scope,5:blkio:/user.slice,2:cpu,cpuacct:/user.slice,1:na"),
+	[]byte("root         104       2 21:34  0.0  0.0     0      0 I<   00:00:00 kthrotld          3 -\nroot         105       2 21:34  0.0  0.0     0      0 S    00:00:00 irq/41-aerdrv     0 -\nroot         107       2 21:34  0.0  0.0     0      0 I<   00:00:00 DWC Notificatio   3 -\nroot         109       2 21:34  0.0  0.0     0      0 S<   00:00:00 vchiq-slot/0      1 -\nroot         110       2 21:34  0.0  0.0     0      0 S<   00:00:00 vchiq-recy/0      3 -"),
+}
+
+func TestParseProcessList(t *testing.T) {
+	for i, ps := range psOutput {
+		t.Run(fmt.Sprintf("iteration %d", i), func(tt *testing.T) {
+			cd := &containerData{}
+			_, err := cd.parseProcessList("/", true, ps)
+			// checking *only* parsing errors - otherwise /proc would have to be emulated.
```

Comment:
do we have some unit tests that check the actual parsing logic and ensure right data is returned after the parse?

Reply:
Unfortunately not. Loop in `parseProcessList()` besides parsing out put of `ps`:
* reads from cgroup fs
* reads from proc fs

I will try to break it even more so that at least some of the code can be tested.

Reply:
There is new shiny test function just below :)

Reply:
awesome, thanks for adding tests here!